opentelemetry-sdk/src/export/logs/mod.rs

```diff
@@ -2,24 +2,26 @@
 use crate::Resource;
 use async_trait::async_trait;
 use opentelemetry_api::{
-    logs::{LogError, LogRecord, LogResult},
+    logs::{LogRecord, LogResult},
     InstrumentationLibrary,
 };
 use std::{borrow::Cow, fmt::Debug};
 
+/// Describes the result of an export.
+pub type ExportResult = LogResult<()>;
```

Comment:
This doesn't seem like an improvement to me. What's the motivation here?

Reply:
- The SpanExporter returns an ExportResult. Having these two very similar things follow the same patterns is something I see as an improvement when trying to write code that functions as both a SpanExporter and LogExporter.
  - It would be even nicer if they both could use the *same* ExportResult, but that's not something that belongs in this change.
- The LogExporter ExportResult as-is is unreferenced.
- The Exporter should probably return an ExportResult. That's what the names imply.

Reply:
Personally, I find `Error` and `Result` type aliases to be a net negative when reading code. IMO they make it harder to understand the actual types that are being passed around by wrapping them in a layer that's hard to pierce for humans (have to look the import or alias definition) while being almost completely transparent to the compiler (no type safe distinction). As such, I feel even stronger about having type aliases of type aliases.

On top of that, giving different types the same name seems to make the type aliases even more confusing for humans.

I understand the goal of consistency between different areas of the code, but if there are inevitable differences between certain mostly-similar components then in my mind we should actually make the differences stand out more rather than obfuscating them.

Reply:
Perhaps this is something we could add to contributing.md because things like this are a bit "style guide"-esque

Reply:
> if there are inevitable differences between certain mostly-similar components then in my mind we should actually make the differences stand out more rather than obfuscating them.

Setting aside the particulars of this precise change, from my perspective as an outside developer writing against an unfamiliar SDK, differences within an SDK are confusing and can be frustrating. Internal consistency (be it patterns, naming, formatting, style, tone, etc.) and predictability aid in helping new users internalize the SDK and use it as a familiar tool in their toolbox, so they can focus on building what they need to rather than focus on the tools themselves.

Reply:
Also, I'd argue that since LogResult<T> is generic, but T must always be () when being returned from something implementing the exporter trait, then a type alias that always provides the correct generic type is beneficial. Every use of LogResult has the same generic type parameter though, so maybe it shouldn't be generic at all.